src/main.cpp

```diff
@@ -396,28 +394,37 @@ int main (int argc, char *argv[]) {
 
   //if (use_sound) Backend_Sound_Pause();
 
-  uint64 framerateMicroseconds = (1000000.0 / FRAMERATE_TARGET) / 2.0;
+  long updatePeriod_nsec = (1000000000.0L / FRAMERATE_TARGET);
 
   gamehacks_init();
 
   /* emulation loop */
   #ifdef __EMSCRIPTEN__
    emscripten_set_main_loop(&mainloop, 60, 1);
   #else
-    uint32 timeNext = 0;
-    // uint32 timePrev = 0;
+    timespec timeAfter, timeBefore;
+
     while(running) {
```

Comment:
Will add a way to use vsync for mainloop timing and fall back to this on refresh rates other than ~60Hz.